### ISSUE TITLE
OKEX CancelOrder error 10000

### ISSUE THREAD

**Issue by gerardo15** [ccxt/ccxt#1082]
Hello.

Whenever i try to cancel order using

await plataforma.cancelOrder(row.uuid_buy, row.mercado)

i get an result false error 10000

Is anyone experience this? i can buy, fetch, but no cancel?

**Comment by kroitor** [comment]
Hi! Can you post your code and **verbose** output?

https://github.com/ccxt/ccxt/wiki/Manual#troubleshooting

**Comment by gerardo15** [comment]
static async cancel(row){
        let plataforma = await this.authorize(row);
        let result = {};
        await plataforma.cancelOrder(row.uuid_buy, okex.fixmercado(row.mercado))
            .then(function(value) {
                console.log(value)
            }, function(reason) {
                console.log(reason)
            });
        return result;
    }


{ Error: okex {"error_code":10000,"result":false}
    at okex.request (C:\Users\lenovo\PhpstormProjects\multi-robots\node_modules\ccxt\js\okcoinusd.js:621:23)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7) constructor: [Function: ExchangeError] }

**Comment by kroitor** [comment]
@gerardo15 this does not look like ccxt code. We don't have the `authorize()` method. Can you show how you instantiate it? When I say "show your code" I mean a minimal snippet that I can copy-paste to reproduce your problem. No need for your keys, we just need to see if you use any rate-limiting, because a cancelOrder can fail if you're hitting a rate limit after createOrder, so the sequence of events really matters, therefore we need to see how it works from the start.

**Comment by gerardo15** [comment]
static async authorize( row ){
        let plataforma = new ccxt.okex();
        plataforma.apiKey = row.apikey;
        plataforma.secret = row.apisecret;
        return plataforma;
    }

then i just call cancel(row)

im doing single calls just to test, i have binance and bittrex working with my code.

**Comment by kroitor** [comment]
```JavaScript
static async authorize( row ){
    let plataforma = new ccxt.okex();
    plataforma.apiKey = row.apikey;
    plataforma.secret = row.apisecret;
    plataforma.verbose = true; // ←-- this is in the Manual, I asked for this, *read the Manual carefully*
    return plataforma;
}

static async cancel(row){
    let plataforma = await this.authorize(row);
    let result = {};
    await plataforma.cancelOrder(row.uuid_buy, row.mercado)
// no need for these lines, they're masking the error reason
//        .then(function(value) {
//            console.log(value)
//        }, function(reason) {
//            console.log(reason)
        });
    return result;
}
```

↑ What do you see after adding the above changes?

**Comment by gerardo15** [comment]
Oh i'm sorry.

okex POST https://www.okex.com/api/v1/cancel_order.do
Request:
 { 'Content-Type': 'application/x-www-form-urlencoded' } api_key=67581183-2f32-****-853f-************&order_id=65885448&sign=D1D8928499AC6B10BC7E7B0AA2A1D6E2
okex POST https://www.okex.com/api/v1/cancel_order.do 200 OK
Response:
{"error_code":10000,"result":false}
(node:11180) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: okex {"error_code":10000,"result":false}
(node:11180) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with
a non-zero exit code.

**Comment by kroitor** [comment]
@gerardo15 add this:

```JavaScript
static async cancel(row){
    let plataforma = await this.authorize(row);
    let result = {};
    console.log ('SELECTED MARKET: ', row.mercado); // ←------- add some logging here
    await plataforma.cancelOrder(row.uuid_buy, row.mercado)
    return result;
}
```

What is the SELECTED MARKET in your case?

**Comment by gerardo15** [comment]
The output is "SELECTED MARKET : DASH/BTC"

**Comment by kroitor** [comment]
Ok, there was an issue with cancelOrder for the OK\* family, now fixed in current version of ccxt 1.10.571. Can you update your version and report back if it now works normally for you or not?
